freeipa/src/main/java/com/sequenceiq/freeipa/service/cloud/CloudResourceRetrievalService.java

```diff
@@ -0,0 +1,40 @@
+package com.sequenceiq.freeipa.service.cloud;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+import com.sequenceiq.cloudbreak.cloud.service.ResourceRetrieval;
+import com.sequenceiq.common.api.type.CommonStatus;
+import com.sequenceiq.common.api.type.ResourceType;
+import com.sequenceiq.freeipa.entity.Resource;
+import com.sequenceiq.freeipa.service.resource.ResourceService;
+
+@Component
+public class CloudResourceRetrievalService implements ResourceRetrieval {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CloudResourceRetrievalService.class);
+
+    @Inject
+    @Qualifier("conversionService")
+    private ConversionService conversionService;
+
+    @Inject
+    private ResourceService resourceService;
+
+    @Override
+    public Optional<CloudResource> findByResourceReferenceAndStatusAndType(String resourceReference, CommonStatus status, ResourceType resourceType) {
```

Comment:
1) In the persister class there is a persist() method. It always returns null. By introducing a retrieving mechanism it might make sense to eliminate that method.